### DISCUSSION TITLE
How to use nested route less layout the right way

### DISCUSSION THREAD

**Discussion by pkishorez** [remix-run/remix#3014] [Q&A] [answered]
I'm trying to create a blog page, such that index has a layout that is different from other nested pages within it.
The route structure I've tried for the same is:
<img width="294" alt="image" src="https://user-images.githubusercontent.com/5671019/165809485-57d04d2e-2569-40df-9dbf-8af2d94e27ed.png">
But that doesn't seem to work as expected. since route /blog is resolving with layout within __nested.

The below routes seem to reflect what I'm aiming for, but the folder structure isn't really clean in my opinion.
<img width="294" alt="image" src="https://user-images.githubusercontent.com/5671019/165808923-6b8f88bc-ecef-484a-b9d7-037f22001b1e.png">

Please help me understand more about how we can use nested route less layouts the right way :)

Thanks in advance.

**Comment by silvenon** [comment, marked as answer]
Another hacky way to achieve this would be:

```
routes
├── blog
│   ├── __home
│   │   └── index.tsx
│   ├── __nested
│   │   ├── $id.tsx
│   │   └── new.tsx
│   └── __nested.tsx
└── index.tsx
```

However, it should work like in your first screenshot, I'm sad that it doesn't (tried it myself), and I remember finding myself in similar trouble. I hope that this is just a temporary problem in Remix, and I recommend raising an issue about this if there isn't one already.

**Reply by pkishorez** [reply]
Thanks a lot. This worked!
And way cleaner than the approach I've taken 👍🏻

**Reply by silvenon** [reply]
@pkishorez apparently this has been fixed in `v1.17.0` https://github.com/remix-run/remix/discussions/3014#discussioncomment-5993007. (I haven't tried it.)

**Reply by brophdawg11** [reply]
Just FYI `1.17.0` is not released yet but when it is it should contain the fix.  You can try via the nightly release from last night though: `v0.0.0-nightly-33cc4c0-20230525`

**Comment by kiliman** [comment]
I believe it's a bug in Remix. The file layout is simply used to create the config needed by `react-router`.

Here's how Remix is seeing the above routes
```ts
<Routes>
  <Route file="root.tsx">
    <Route path="blog/" file="routes/blog/__nested.tsx">
      <Route path=":id" file="routes/blog/__nested/$id.tsx" />
      <Route path="new" file="routes/blog/__nested/new.tsx" />
    </Route>
    <Route path="blog" index file="routes/blog/index.tsx" />
    <Route index file="routes/index.tsx" />
  </Route>
</Routes>
```
When it should be like this:
```ts
<Routes>
  <Route file="root.tsx">
    <Route path="blog/">
      <Route index file="routes/blog/index.tsx" />
      <Route file="routes/blog/__nested.tsx">
        <Route path=":id" file="routes/blog/__nested/$id.tsx" />
        <Route path="new" file="routes/blog/__nested/new.tsx" />
      </Route>
    </Route>
    <Route index file="routes/index.tsx" />
  </Route>
</Routes>
```

**Reply by pkishorez** [reply]
True. Index route should take priority and then the more generic ones like `$id`.

**Reply by silvenon** [reply]
Thanks so much for confirming this @kiliman, I'll go raise an issue if nobody beats me to it. Btw, I think there's a small mistake in your "expected" code block, I assume the `routes/blog/index.tsx` and `routes/index.tsx` route elements were meant to be switched.

**Reply by kiliman** [reply]
Ha. Probably, I was copy+pasting. I'll fix it.

**Reply by silvenon** [reply]
Bug report is here: #3052

**Reply by brophdawg11** [reply]
Thanks for the bug report!  I'm looking into this in https://github.com/remix-run/remix/pull/3986

**Reply by brophdawg11** [reply]
ok sorry for the delay 😬 , but I just merged a fix for this in #4355 which should get included in the next release

**Reply by silvenon** [reply]
Awesome, thank you so much! 🎉

**Reply by brophdawg11** [reply]
Sorry folks we found a small issue with that approach so reverted and re-addressing in https://github.com/remix-run/remix/pull/4421

**Reply by brophdawg11** [reply]
Sorry for the delay!  #4421 is now merged and should be available once `v1.17.0` is released
